R/slice_derivation.R

```diff
@@ -0,0 +1,171 @@
+#' Execute a Derivation with Different Arguments for Subsets of the Input Dataset
+#'
+#' The input dataset is split into slices (subsets) and for each slice the
+#' derivation is called separately. Some or all arguments of the derivation
+#' may vary depending on the slice.
+#'
+#' @param dataset Input dataset
+#'
+#' @param derivation Derivation
+#'
+#' @param args Arguments of the derivation
+#'
+#'   A `param()` object is expected.
+#'
+#' @param ... A `derivation_slice()` object is expected
+#'
+#'   Each slice defines a subset of the input dataset and some of the parameters
+#'   for the derivation. The derivation is called on the subset with the
+#'   parameters specified by the `args` parameter and the `args` field of the
+#'   `derivation_slice()` object. If a parameter is specified for both, the
+#'   value in `derivation_slice()` overwrites the one in `args`.
+#'
+#' @details
+#'
+#'   For each slice the derivation is called on the subset defined by the
+#'   `filter` field of the `derivation_slice()` object and with the parameters
+#'   specified by the `args` parameter and the `args` field of the
+#'   `derivation_slice()` object. If a parameter is specified for both, the
+#'   value in `derivation_slice()` overwrites the one in `args`.
+#'
+#'   - Observations that match with more than one slice are only considered for
+#'   the first matching slice.
+#'
+#'   - Observations with no match to any of the slices are included in the
+#'   output dataset but the derivation is not called for them.
+#'
+#' @return The input dataset with the variables derived by the derivation added
+#'
+#' @keywords user_utility high_order_function
+#'
+#' @author Stefan Bundfuss
+#'
+#' @seealso [params()] [restrict_derivation()]
+#'
+#' @export
+#'
+#' @examples
+#'
+#'  library(stringr)
+#'  advs <- tibble::tribble(
+#'   ~USUBJID, ~VSDTC,       ~VSTPT,
+#'   "1",      "2020-04-16", NA_character_,
+#'   "1",      "2020-04-16", "BEFORE TREATMENT"
+#' )
+#'
+#' # For the second slice no filter is specified. Thus derive_vars_dtm is called
+#' # with time_imputation = "last" for all observations which do not match for the
+#' # first slice.
```

Comment:
Can we make this somehow more explicit? I feel like just not specifiying the `filter` parameter is confusing...

Reply:
Updated